JsonPath/JsonPath.csproj

```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		<TargetFrameworks>netstandard2.1;net8.0</TargetFrameworks>
+		<TargetFrameworks>netstandard2.1;net8.0;netstandard2.0</TargetFrameworks>
```

Comment:
I'd prefer just replacing `netstandard2.1`.

Reply:
Tried removing netstandard2.1 target however this causes the build to fail... which I believe is due to the [NotNullWhen](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute?view=netstandard-2.1) only being available in netstandard2.1 and beyond? Could be wrong though doesn't really make sense to me the error?

```JsonPath.Tests\Suite\CburgmerFeatureValidationTests.cs(139,54,139,65): error CS0433: The type 'NotNullWhenAttribute' exists in both 'JsonPath.Net, Version=0.8.0.0, Culture=neutral, PublicKeyToken=17ed63f672b0e278' and 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'```

Reply:
I wouldn't expect it to build against netstandard2.0, then.

Make sure that PolySharp is added to the project. There's a reference and a project property.  Check the other projects for examples.

Reply:
The error relates to the test project, not the package library project.

I think it is getting confused with the multi targeting from a test perspective - It does build & tests all pass when all 3 target's are present. I presume when `netstandard2.1` is targeted, the tests are able to pull through the attribute because of that.. Not too sure if i'm honest.

PolySharp is added already to JsonPath project & csproj property is present.
`<PolySharpIncludeRuntimeSupportedAttributes>true</PolySharpIncludeRuntimeSupportedAttributes>`
`<PackageReference Include="PolySharp" Version="1.14.1" PrivateAssets="All" />`

Reply:
@Sergio0694 do you know what's going on here?  I'd expect `[NotNullWhenTrue]` to be added for `netstandard2.0`.

Reply:
Oh!  It's because internals are visible to the tests project.

PolySharp is generating the attribute in _JsonPath.Net_ as an internal type in .Net Standard 2.0.  Then that version is being pulled for the .Net 6 version of the tests.  But since .Net 6 already defines the attribute, there's an ambiguity.

Reply:
@Emcrank just remove the attribute for now, please.  I'll figure it out later.

Reply:
@gregsdennis Okay thanks, will do. Not familiar with the workings of PolySharp but makes sense.

Reply:
That's committed.